### ISSUE TITLE
<W> fish: Locking the universal var file took too long (1.001 seconds).

### ISSUE THREAD

**Issue by mec07** [acomagu/fish-async-prompt#3]
Awesome plugin. I really like it. There are a couple of issues. If you hit enter several times you get this error message (after which it blocks for a little bit): 
`<W> fish: Locking the universal var file took too long (1.001 seconds).`

Also it ignores new lines in my fish prompt :(

Thanks for making it!

**Comment by mec07** [comment]
Some extra info for you: 
`fish, version 2.7.1`
Mac OSX 10.13.6
Iterm2 Build 3.2.3

**Comment by acomagu** [comment]
Thank you for feedback. Please tell me some more.
- How do you define your prompt function?
- Any configuration for this plugin?
- The output of `set -U | grep __async_prompt`

**Comment by acomagu** [comment]
I fixed a bug temporary universal variables are not removed so could you update the plugin and try again?

**Comment by mec07** [comment]
Hi,

I've looked into it a little more and I've found an autogenerated file called `~/.config/fish/fishd.f07960158f86` which has the following contents:
```
# This file is automatically generated by the fish.
# Do NOT edit it directly, your changes will be overwritten.
SET __async_prompt_fish_prompt_text_26996:\x1b\x5b97m\x1b\x5b46m\u22ca\x3e\x1b\x5b30m\x1b\x28B\x1b\x5bm\x20\x1b\x5b33m\x7e/d/thesis\x1b\x5b30m\x1b\x28B\x1b\x5bm\x20on\x20\x1b\x5b32mmaster\x1b\x5b30m\x1b\x28B\x1b\x5bm\x20\u2a2f\x20\x20\x1b\x5b34m\x28london\x29\x1e\x3e\x20
SET __async_prompt_fish_prompt_text_32278:\x1b\x5b97m\x1b\x5b46m\u22ca\x3e\x1b\x5b30m\x1b\x28B\x1b\x5bm\x20\x1b\x5b33m\x7e/d/thesis\x1b\x5b30m\x1b\x28B\x1b\x5bm\x20on\x20\x1b\x5b32mmaster\x1b\x5b30m\x1b\x28B\x1b\x5bm\x20\u2a2f\x20\x20\x1b\x5b34m\x28draco\x29\x1e\x3e\x20
SET __async_prompt_fish_right_prompt_text_32278:\x1d
SET __fish_init_2_39_8:\x1d
SET __fish_init_2_3_0:\x1d
SET fish_color_autosuggestion:555\x1ebrblack
SET fish_color_command:\x2d\x2dbold
SET fish_color_comment:red
SET fish_color_cwd:green
SET fish_color_cwd_root:red
SET fish_color_end:brmagenta
SET fish_color_error:brred
SET fish_color_escape:bryellow\x1e\x2d\x2dbold
SET fish_color_history_current:\x2d\x2dbold
SET fish_color_host:normal
SET fish_color_match:\x2d\x2dbackground\x3dbrblue
SET fish_color_normal:normal
SET fish_color_operator:bryellow
SET fish_color_param:cyan
SET fish_color_quote:yellow
SET fish_color_redirection:brblue
SET fish_color_search_match:bryellow\x1e\x2d\x2dbackground\x3dbrblack
SET fish_color_selection:white\x1e\x2d\x2dbold\x1e\x2d\x2dbackground\x3dbrblack
SET fish_color_user:brgreen
SET fish_color_valid_path:\x2d\x2dunderline
SET fish_greeting:\x1d
SET fish_key_bindings:fish_default_key_bindings
SET fish_pager_color_completion:\x1d
SET fish_pager_color_description:B3A06D\x1eyellow
SET fish_pager_color_prefix:white\x1e\x2d\x2dbold\x1e\x2d\x2dunderline
SET fish_pager_color_progress:brwhite\x1e\x2d\x2dbackground\x3dcyan
SET fish_user_paths:/Users/mcoury/\x2efzf/bin
```
It seems like something is going slightly wrong...

**Comment by mec07** [comment]
Actually when I open up a new terminal now I no longer have a prompt, I only have the error! 
```
cat: /proc/self/stat: No such file or directory
fish: option requires an argument -- c
cat: /proc/self/stat: No such file or directory
fish: option requires an argument -- c
```
I can enter commands, but I don't see a prompt anymore.

**Comment by mec07** [comment]
I've made a PR to fix this issue: #4 Please review when you get a moment :)

**Comment by acomagu** [comment]
Thank you for PR!
I'm really sorry for breaking shell by an other change. I don't have any macOS PC so it's really hard to test...

> Unfortunately I still seem to be getting <W> fish: Locking the universal var file took too long (0.577 seconds).
Hmm. I'll try to reproduce the problem with your configuration file.
Thank you for giving it.

**Comment by acomagu** [comment]
I created another branch has only related changes: [warn-looking-var](https://github.com/acomagu/fish-async-prompt/tree/warn-looking-var)
On this branch, the problem missing `/proc/stat` won't occur.

**Comment by mec07** [comment]
Thanks. I've had a look at the fish github and I think you'll find this interesting: https://github.com/fish-shell/fish-shell/blob/786c0c5abb936eeb10eae45b508132a313bc8abc/CHANGELOG.md#notable-non-backward-compatible-changes

Perhaps we could just make the function check the new format `$fish_pid` and if that is empty then return the old format `%self`. I'll update the PR.

**Comment by acomagu** [comment]
I merged #4. Thanks a lot!
